test/test-pre-push.sh

```diff
@@ -2,6 +2,85 @@
 
 . "test/testlib.sh"
 
+begin_test "pre-push with good ref"
+(
+  set -e
+  reponame="pre-push-master-branch-required"
+  setup_remote_repo "$reponame"
+  clone_repo "$reponame" "$reponame"
+
+  git config "lfs.$(repo_endpoint "$GITSERVER" "$reponame").locksverify" false
+  git lfs track "*.dat"
+  echo "hi" > a.dat
+  git add .gitattributes a.dat
+  git commit -m "add a.dat"
+
+  refute_server_object "$reponame" 98ea6e4f216f2fb4b69fff9b3a44842c38686ca685f3f55dc48c5d3fb1107be4 "refs/heads/master"
+
+  # for some reason, using 'tee' and $PIPESTATUS does not work here
+  echo "refs/heads/master master refs/heads/master 0000000000000000000000000000000000000000" |
+    git lfs pre-push origin "$GITSERVER/$reponame" 2>&1 > push.log
+
+  assert_server_object "$reponame" 98ea6e4f216f2fb4b69fff9b3a44842c38686ca685f3f55dc48c5d3fb1107be4 "refs/heads/master"
+)
+end_test
+
+begin_test "pre-push with tracked ref"
+(
+  set -e
+  reponame="pre-push-tracked-branch-required"
+  setup_remote_repo "$reponame"
+  clone_repo "$reponame" "$reponame"
+
+  git config "lfs.$(repo_endpoint "$GITSERVER" "$reponame").locksverify" false
+  git lfs track "*.dat"
+  echo "hi" > a.dat
+  git add .gitattributes a.dat
+  git commit -m "add a.dat"
+
+  refute_server_object "$reponame" 98ea6e4f216f2fb4b69fff9b3a44842c38686ca685f3f55dc48c5d3fb1107be4 "refs/heads/tracked"
+
+  # for some reason, using 'tee' and $PIPESTATUS does not work here
```

Comment:
I believe that `${PIPESTATUS[1]} would work, since there are two commands being piped here.

Reply:
That must be copy pasta, I don't recall any issues. At any rate, it's just a test so it doesn't matter ;)